Source/VungleAdapter.swift

```diff
@@ -48,30 +40,27 @@ final class VungleAdapter: PartnerAdapter {
         log(.setUpStarted)
         
         // Get credentials, fail early if they are unavailable
-        guard let appId = configuration.credentials[.appIDKey] as? String, !appId.isEmpty else {
+        guard let appID = configuration.credentials[.appIDKey] as? String, !appID.isEmpty else {
             let error = self.error(.initializationFailureInvalidCredentials, description: "Missing \(String.appIDKey)")
             self.log(.setUpFailed(error))
             completion(error)
             return
         }
         
-        // Vungle provides one single delegate for all ads.
-        // VungleAdapterRouter implements these delegate protocols and forwards calls to the corresponding partner ad instances.
-        let router = VungleAdapterRouter(adapter: self)
-        self.router = router    // keep the router instance alive
-        VungleSDK.shared().delegate = router
-        VungleSDK.shared().sdkHBDelegate = router
-        
-        // Disable banner auto-refresh for all Vungle ads. Auto-refresh is handled by Chartboost Mediation.
-        VungleSDK.shared().disableBannerRefresh()
-        
-        setUpCompletion = completion
-        
+        // Apply saved COPPA setting before init, as suggested in documentation
+        // https://support.vungle.com/hc/en-us/articles/360048572411#recommendations-for-using-vungle-s-coppa-compliance-tools-0-6
+        let savedCOPPASetting = UserDefaults.standard.bool(forKey: COPPA_KEY)
+        VunglePrivacySettings.setCOPPAStatus(savedCOPPASetting)
         // Initialize Vungle
-        do {
-            try VungleSDK.shared().start(withAppId: appId)
-        } catch {
-            vungleSDKFailedToInitializeWithError(error)
+        VungleAds.initWithAppId(appID) { initError in
```

Comment:
nit: maybe avoid keeping a strong reference to self in this closure

Reply:
[done](https://github.com/ChartBoost/chartboost-mediation-ios-adapter-vungle/pull/44/commits/be1d0dea13dd0e7377b51dbc44b10af4c9ac5040)